src/mlpack/methods/ann/ffn.hpp

```diff
@@ -142,6 +142,16 @@ class FFN
   void Predict(arma::mat& predictors, arma::mat& results);
 
   /**
+   * Predict the responses to a given set of predictors. The responses will
+   * reflect the output of the given output layer as returned by the
+   * output layer function.
+   *
+   * @param predictors Input predictors.
+   * @return Output predictions of responses into.
+   */
+  arma::mat Predict(arma::mat predictors);
+
```

Comment:
Why not add two overloads:

```
void Predict(const mat& predictors, mat& output);
void Predict(mat&& predictors, mat& output);
```

This would allow you to avoid having two dissimilar calls that do the same thing, and it's easy to specify when you want the original matrix unchanged and when you don't.